en/guide/error-handling.md

```diff
@@ -42,6 +42,49 @@ app.get('/', function (req, res, next) {
 })
 ```
 
+Starting with Express 5, route handlers and middleware that return a Promise will call 
+`next(value)` automatically when they reject or throw an error.  For example:
+
+```js
+app.get('/user/:id', function (req, res, next) {
+  return getUserById(req.params.id).then(function (user) {
+    res.send(user)
+  })
+})
+```
+
+If `getUserById` throws an error or rejects, `next` will be called with either the thrown error or the rejected value.  
+If no rejected value is provided, `next` will be called with the string `Rejected promise`.
+
+If you want to handle the error before passing it along to `next`, you can use a try/catch block (for functions using async/await) or add a `.catch` to your Promise (or an onRejected callback to `.then`).  For example:
+
+```js
+app.get('/user/:id', function (req, res, next) {
+  return getUserById(req.params.id)
+    .then(function (user) {
+      res.send(user)
+    })
+    .catch(function (error) {
+    // handle the error
+      next(error)
+    })
+})
+```
+
+With async/await:
+
+```js
+app.get('/user/:id', async function (req, res, next) {
+  try {
+    var user = await getUserById(req.params.id)
+    res.send(user)
+  } catch (error) {
+    // handle the error
+    next(error)
+  }
+})
+```
```

Comment:
I think what's most important in this doc is that rejected promises automatically call next with an error. 

So for the async await case, catching the error like this isn't really any different from how things would be done in v4. The real difference is in a case like this:

```js
app.get('/user/:id', async function (req, res, next) {
  // simulating an uncaught rejection
  await Promise.reject("BOOM")
  res.send("ok")
})
```

In v4, `res.send` would never be called, and the error wouldn't be handled. In v5, `res.send` is still not called, but Express will catch this error and forward it to an error handler.

Reply:
thanks @jonchurch!  I'll update this section